src/Microsoft.OData.Core/JsonLight/ODataJsonLightPropertyAndValueDeserializer.cs

```diff
@@ -1523,8 +1523,25 @@ private object ReadEnumValue(bool insideJsonObjectValue, IEdmEnumTypeReference e
                 throw new ODataException(ODataErrorStrings.JsonReaderExtensions_UnexpectedNodeDetectedWithPropertyName(JsonNodeType.PrimitiveValue, JsonNodeType.StartObject, propertyName));
             }
 
-            string enumStr = this.JsonReader.ReadStringValue();
-            return new ODataEnumValue(enumStr, expectedValueTypeReference.FullName());
+            object enumValue = this.JsonReader.ReadPrimitiveValue();
+            string enumString;
+            if (enumValue is int enumIntValue)

```

Comment:
>enumValue is int enumIntValue [](start = 16, length = 29)

This new syntax maybe not pass .NET 4.5 and .NET 3.5 compiler. Please change it to normal.